agent/testflinger_agent/runner.py

```diff
@@ -78,16 +80,16 @@ def post_output(self, data: str):
     def register_stop_condition_checker(self, checker: StopConditionType):
         self.stop_condition_checkers.append(checker)
 
-    def check_stop_conditions(self) -> str:
+    def check_stop_conditions(self) -> Tuple[Optional[TestEvent], str]:
         """
         Check stop conditions and return the reason if any are met. Otherwise,
         return an empty string if none are met
         """
         for checker in self.stop_condition_checkers:
             output = checker()
-            if output:
+            if output[0] is not None:
```

Comment:
If we're going to do it this way, then I think we shouldn't just overload output to make it a tuple. Instead, maybe we could get `event, detail` = checker()` then you could specifically check `if event is not None:`...